src/redecomp/transfer/SparseMatrixTransfer.hpp

```diff
@@ -0,0 +1,120 @@
+ * mfem::ParMesh.
+ *
+ * The class currently supports order 0 L2 fields (one value per element).
+ */
+class SparseMatrixTransfer
+{
+public:
+  /**
+   * @brief Construct a new SparseMatrixTransfer object
+   * 
+   * @param parent_test_space Test finite element space on parent mfem::ParMesh
+   * @param parent_trial_space Trial finite element space on parent mfem::ParMesh
+   * @param redecomp_test_space Test finite element space on RedecompMesh
+   * @param redecomp_trial_space Trial finite element space on RedecompMesh
+   */
+  SparseMatrixTransfer(
+    const mfem::ParFiniteElementSpace& parent_test_space,
+    const mfem::ParFiniteElementSpace& parent_trial_space,
+    const mfem::FiniteElementSpace& redecomp_test_space,
+    const mfem::FiniteElementSpace& redecomp_trial_space
+  );
+
+  /**
+   * @brief Transfers RedecompMesh sparse matrix to parent mfem::ParMesh hypre par matrix
+   *
+   * @param src Sparse matrix whose rows and columns are linked to test and trial FiniteElementSpaces on the
+   * RedecompMesh
+   * @return Owned pointer to mfem::HypreParMatrix on the parent mesh
+   *
+   * @note The HypreParMatrix is on the global dofs.  Apply mfem::RAP to reduce to tdofs.
+   */
+  std::unique_ptr<mfem::HypreParMatrix> TransferToParallel(const mfem::SparseMatrix& src) const;
+
+private:
+  /**
+   * @brief Get the RedecompMesh from the finite element space
+   *
+   * This method verifies the finite element space does have a RedecompMesh before returning. Finite element spaces with
+   * no RedecompMesh trigger a SLIC_ERROR macro.
+   * 
+   * @param fe_space Finite element space linked to a RedecompMesh
+   * @return const RedecompMesh& Redecomp mesh associated with finite element space
+   */
+  static const RedecompMesh& getRedecompMesh(const mfem::FiniteElementSpace& fe_space);
+
+  /**
+   * @brief Build vector of parent element offsets for each MPI rank
+   * 
+   * @param redecomp_mesh RedecompMesh whose parent linked mesh the element offset vector is desired
+   * @return axom::Array<HYPRE_BigInt> Array of element offsets for each MPI rank
+   */
+  static axom::Array<HYPRE_BigInt> BuildParentElementRankOffsets(const RedecompMesh& redecomp_mesh);
+
+  /**
+   * @brief Returns MPIUtility pointer for the MatrixTransfer object
+   * 
+   * @return const MPIUtility* 
+   */
+  const MPIUtility& getMPIUtility() const;
+
+  /**
+   * @brief Test finite element space on the parent mfem::ParMesh
+   */
+  const mfem::ParFiniteElementSpace& parent_test_space_;
+
+  /**
+   * @brief Trial finite element space on the parent mfem::ParMesh
+   */
+  const mfem::ParFiniteElementSpace& parent_trial_space_;
+
+  /**
+   * @brief Test finite element space on the RedecompMesh
+   */
+  const mfem::FiniteElementSpace& redecomp_test_space_;
+
+  /**
+   * @brief Trial finite element space on the RedecompMesh
+   */
+  const mfem::FiniteElementSpace& redecomp_trial_space_;
+
+  /**
+   * @brief Test RedecompMesh
```

Comment:
Not strictly required, but consider a note somewhere indicating that since you support rectangular sparse matrices you need to have two redecomp meshes. Up to now the paradigm has only been one on-rank redecomp mesh for a given redecomposition. It wouldn't hurt elaborating on why, even if internally, two are required.